### ISSUE TITLE
LMMS Crowdfunding Project

### ISSUE THREAD

**Issue by Snakezer** [LMMS/lmms#3794]
After having a discussion and proposing an idea in #2745 and on the LMMS Discord Server.I have decided to open up a issue for the proposal.
The proposal requires the following things,to be executed properly:-

- Developers who understand the core codebase to write documentation on the developer wiki.Specially,the architecture page(https://github.com/LMMS/lmms/wiki/LMMS-Architecture).
(@PhysSong has already agreed to work on this,more people needed.)

- All of the goals,prices and timelines of the crowdfunding project need to be identified,before starting a campaign. (This will have to be decided by developers while getting feedback from users.)

- People will be required to find and contact Qualified C++ Developers. ( I might do part of this but more people will be required to find new developers.)

- People would be required to market and manage the campaign.(I am ready to work on this but any help would be really appreciated).

- People would be required to make kick-backs for donors. _i.e.-early ready-to-run alpha builds,instrument presets,sample packs and tutorials._( @cubician has agreed to make extra tutorials for the project.However,more people would be needed to make presets,record/donate samples and build alpha code).

- A secure place needs to be decided upon to keep funds from the campaign. ( @tresf is currently in-charge of funds for LMMS, but,a bank account might be needed to be created).

- The 1.2.0 milestone (https://github.com/LMMS/lmms/milestone/3) will need to be completed,and LMMS 1.2.0. will need a release as the crowdfunding project will not affect the development of 1.2.0. ,in any way possible.

> TL;DR:- A lot of work will need to be completed before a crowdfunding campaign can be initiated. However,this can drastically affect the development speed and features LMMS exhibits in the future.Thus,Anybody who can help out with anything in regard to this project,is requested to do so.

**Comment by musikBear** [comment]
imo a guarantee of not selling of commercializing the project is a prerequist before this future project 2.0? is initiated -Maybe just one of my typicals annoying self-obvious mehehs, but i have seen Gamemaker go that way down the drain, even after promises given! So i have prios belly-punching in the history of open and free software!

**Comment by tresf** [comment]
@musikBear Commercializing LMMS isn't on the roadmap currently, so let's try to keep speculation aside of @Snakezer's thread so that he can use this to triage concrete steps towards a campaign.

Crowdfunding and commercializing are two quite separate tasks.  If LMMS were to go commercial, it would certainly stir a lot of commotion but it would also require a good business plan.  Crowdfunding is a one-off shot at something the project couldn't have afforded otherwise.  For example, in 2013, [VLC crowdfunded it's ModernUI rewrite](https://www.kickstarter.com/projects/1061646928/vlc-for-the-new-windows-8-user-experience-metro) yet it's still free. :)

**Comment by fundamental** [comment]
Don't forget taxes on the crowdfunding list. If you hit the point of receiving (non-trivial amounts of) money through one person and paying multiple other people in different countries taxes (and organizational status) become rather complex.

**Comment by tresf** [comment]
@fundamental unless you're talking about officers receiving company funds in several countries, as far as I understand the US makes it pretty simply.  [Hired work is simply a company expense](https://money.stackexchange.com/a/35658).  The tax burden is on the recipient, not on the organization.

**Comment by miketurn** [comment]
test

**Comment by miketurn** [comment]
Obviously this proposal is in its beginning stages so I don't expect these questions to have answers, but I have a couple of questions.  (Also keep in mind that I don't know a ton about crowdfunding)

1.) Will the crowd funding go just towards larger projects that involve complete new coding, or will it also go towards things such as minor updates / bug fixes?

2A.) Does this crowd funding mean that the person who works on and completes a specific task gets paid the funding raised for that said project?  
2B.) If so what if many people work on a project together?
2C.) What about other people (non-coders) involved such as the person who comes up the idea and or request / person who creates graphics / etc?

3.) This will sound weird, obviously the more funding an idea gets, the more desired it is, but what if a lot of funding is raised for a feature / GUI update / etc. that isn't the best for the program itselt.  Will there be someone to oversee ideas and make a decision fore or against it, even if it has the funding to back it up?

4.) In the proposal it states....
"People will be required to find and contact Qualified C++ Developers"
To my understanding, LMMS uses QT5 coding, which I thought was its own language.
Can someone explain what this request is for?

5.) I get a little confused at times about the "free as in....  blah blah" :)
Some free features / formats / programs / coding / etc. that are offered to developers to be used within programs such as LMMS for free, would anything about crowdfunding cause anything not to be able to be added to LMMS because now funding is being raise? (Hopefully this makes sense) 

Overall though, if generally everything goes smooth with this idea and it doesn't change LMMS drastically in a different direction, as stated above this could be a very beneficial thing for the future of LMMS.

**Comment by miketurn** [comment]
6.) What about items that will benefit LMMS but don't get the funding raised?
Will need something put into place so that small little fixes / updates don't get overlooked, due to limited/no funding.

**Comment by tresf** [comment]
> 1.) Will the crowd funding go just towards larger projects that involve complete new coding, or will it also go towards things such as minor updates / bug fixes?

The campaign, --  if formed -- must outline this.  No campaign has been formed at this time.   You're jumping the gun.

> 2A.) Does this crowd funding mean that the person who works on and completes a specific task gets paid the funding raised for that said project?
2B.) If so what if many people work on a project together?
2C.) What about other people (non-coders) involved such as the person who comes up the idea and or request / person who creates graphics / etc?

Stop.  The campaign defines goals.  Who gets paid is subjective to the person running the campaign.  Don't micromanage this.  It's the campaign manager's duty, not yours.  Don't contribute to a campaign manager you don't trust.

> 3.) This will sound weird, obviously the more funding an idea gets, the more desired it is, but what if a lot of funding is raised for a feature / GUI update / etc. that isn't the best for the program itselt. Will there be someone to oversee ideas and make a decision fore or against it, even if it has the funding to back it up?

Over-funding is a threat in every campaign.  Hopefully backers trust the campaign manager, otherwise they're potentially buying snake-oil.

> 4.) In the proposal it states....
"People will be required to find and contact Qualified C++ Developers"
To my understanding, LMMS uses QT5 coding, which I thought was its own language.
Can someone explain what this request is for?

Google Qt5.  It's a C++ framework.  This question is lacking adequate research.  Qt5 knowledge is a prerequisite of the task, just like .NET or Java would be.  Just look it up.

> 5.) I get a little confused at times about the "free as in.... blah blah" :)
Some free features / formats / programs / coding / etc. that are offered to developers to be used within programs such as LMMS for free, would anything about crowdfunding cause anything not to be able to be added to LMMS because now funding is being raise? (Hopefully this makes sense)

Links were provided in the "free as in" references. Read them.

> Overall though, if generally everything goes smooth with this idea and it doesn't change LMMS drastically in a different direction, as stated above this could be a very beneficial thing for the future of LMMS.

This is speculative.  Please use clear arguments, they're more effective. <3

**Comment by tresf** [comment]
To elaborate a bit.... What's important to remember is LMMS uses a source-code license -- GPL -- that guarantees it will be open source forever.

This makes no guarantee (ever, even now) that it will be make available for download for free.  GPL protects source, not binary.  This is why free beer is different than freedom of speech and you need to know this before pitching money at anything.

In regards to source vs. binary -- CentOS and RedHat have played this dance.  RedHat being commercial, CentOS being free to download.  Be defensive and be protective but also get used to it.  This isn't the thread to explain GPL.

**Comment by miketurn** [comment]
@tresf
Thank you for your responses, like I said in the beginning of my post I understand that this concept is in the beginning stages, so I was not expecting full abosolute accurate answers, but again thank you for touching upon each topic, this was helpful.

My post was getting lengthy so I tried to limit the specifics my question #4 about the C++ developers contained.  I imagined that QT5 was built off of C++, just around here usually when members mention coding they refer to specifically the QT language.  (I know you don't like when people do this but) I read something in the past about the future of LMMS, where it was stated that at some point LMMS would reach its limits with the current language it was written in, meaning that certain items can never be added just due to the nature of the language it was created in.
So when C++ developers were mentioned and talks about funding, I just thought that maybe really big changes were coming, like a full language change :)
I apologize for this question.

Thanks again

**Comment by tresf** [comment]
Qt is C++, which really has no limits as a programming language in regards to DSP and hardware performance.

If a port to mobile happens, there may need to be some other languages, but Qt is a C++ API, not a language and it is a very well developed and supported language and is used for much more complicated tasks than processing audio (e.g. see VirtualBox)

**Comment by tresf** [comment]
I don't think there's any argument on how this would be executed but the project still lacks a leader with the free time required to spearhead this initiative.
